crates/uv/src/commands/project/lock.rs

```diff
@@ -41,9 +42,23 @@ pub(crate) async fn lock(
 
     // Find an interpreter for the project
     let interpreter = match project::find_environment(&workspace, cache) {
-        Ok(environment) => environment.into_interpreter(),
+        Ok(environment) => {
+            let interpreter = environment.into_interpreter();
+            if let Some(python) = python.as_deref() {
+                let request = ToolchainRequest::parse(python);
+                if request.satisfied(&interpreter) {
+                    interpreter
+                } else {
+                    Toolchain::find_requested(python, SystemPython::Allowed, preview, cache)?
```

Comment:
I guess I should update this API to take the `ToolchainRequest` instead of a string.